gst/tensor_converter/tensor_converter.c

```diff
@@ -887,16 +910,161 @@ gst_tensor_converter_query_caps (GstTensorConverter * self, GstPad * pad,
 {
   GstCaps *caps;
 
-  if (pad == self->sinkpad) {
+  caps = gst_pad_get_current_caps (pad);
+  if (!caps) {
     caps = gst_pad_get_pad_template_caps (pad);
-  } else {
-    caps = gst_tensor_caps_from_config (&self->tensor_config);
+  }
+
+  if (pad == self->sinkpad) {
+    GstCaps *peer_caps;
+
+    /* get possible caps from downstream element */
+    peer_caps = gst_pad_peer_query_caps (self->srcpad, NULL);
+
+    if (peer_caps) {
+      silent_debug_caps (peer_caps, "peer caps");
+
+      if (gst_caps_get_size (peer_caps) > 0) {
+        GstTensorConfig config;
+        GstStructure *st;
+        GstCaps *media_caps, *tmp;
+        guint i, caps_len;
+        media_type type;
+
+        /* get tensor info from peer caps */
+        st = gst_caps_get_structure (peer_caps, 0);
+        gst_tensor_config_from_structure (&config, st);
+
+        /* convert peer caps to possible media caps */
+        media_caps = gst_caps_from_string (GST_TENSOR_MEDIA_CAPS_STR);
+        caps_len = gst_caps_get_size (media_caps);
+
+        for (i = 0; i < caps_len; ++i) {
+          st = gst_caps_get_structure (media_caps, i);
```

Comment:
https://developer.gnome.org/gstreamer/stable/gstreamer-GstCaps.html#gst-caps-get-structure

To modify ```st```, you need ```gst_caps_make_writable()``` first.